### PR TITLE
feat(report): add emergency command

### LINKED ISSUE #41: [FEAT] Human readable timestamp in report cog.
# Overview
The text in the timestamp field of the report cog could be much more readable.

# Commands
```
[p]suggested commands and usages
```

# DoD
* "Timestamp" field should be human readable. Suggest YYYY/MM/DD - HH:MM

# Stretch
* None.

### LINKED ISSUE #54: [FEAT] Emergency reports
# Overview
## Describe the solution you'd like
Feature partity with `[p]em` command from @the1337g33k's Araquiel bot.
User runs `[p]em <optional message>` in any given channel, the message is deleted, the report is sent to the predefined report channel, and all active moderators are pinged.
## Is this suggestion based on an existing problem? If so, please describe it

# DoD
* `[p]em` or `[p]emergency` command, allowing for an optional message. Same as current report cog functionality, but all active moderators are mentioned.

### PR DESCRIPTION
**Pull request by issy**
# Initial Checklist
- [x] Has @tigattack and/or @kdavis been added as a reviewer?
- [x] If applicable, have the relevant project(s), milestone(s), and label(s) been applied?
- [ ] If applicable, have you added details of the cog to the readme as per [README.md](https://github.com/rHomelab/LabBot-Cogs/blob/develop/README.md#cog-summaries)?

# Details
* **Does this resolve an issue?**
Resolves #54 
Resolves #41 

### New Features
* Adds emergency command. Just like report command, but pings all active mods

### Breaking Changes
* Causes merge conflict with (currently unmerged) PR #87 

## Additional
This also incorporates the changes from (currently unmerged) PR #87, so #87 should not be merged if this PR is to be merged

(I also removed some trailing whitespace from the autoreact cog lol)

### CODE REVIEW

**Comment by the1337g33k** [comment]
One quick tweak, line 79 of report should ping online and idle states.  I know I would like reports even though I sit in idle status 24/7.  Otherwise all looks good!

**Comment by issy** [comment]
> One quick tweak, line 79 of report should ping online and idle states. I know I would like reports even though I sit in idle status 24/7. Otherwise all looks good!

I've looked into when the desktop client goes into the idle state and apparently it's after no activity (mouse movement, keypresses, voice activity) for about 10 minutes. In my opinion the bot shouldn't ping people who are idle, as they're effectively offline (example: as I write this half of the mods are set to idle and I know for a fact that some of those people are sleeping). I'll wait for further feedback to see if there's pushback on this, as there are definitely two sides to it (could be in the same room as your PC but not near it, and you can hear the ping notification sound). IMO if you purposefully set yourself to idle then it's to emulate being afk, and should be treated as such.

**Comment by issy** [comment]
Cog hasn't been tested yet. Would prefer if it was tested on the group test server so we can have a larger sample of statuses

**Comment by issy** [comment]
I considered also pinging DND mods, but I figured that if you don't wanna be disturbed then you shouldn't be disturbed. Something I also thought of is a sort of "status hierarchy. So, if no mods are online, it pings mods in DND, or mods in idle or whatever. If there is call for this then I'll include it, but I could definitely see reasonss for not wanting a feature like this also :)

**Comment by the1337g33k** [comment]
We had it setup this way in Araquiel.  There was a reserve function where if everyone happened to be offline it would then go ahead and ping everyone no matter the status but that was never needed.

**Comment by tigattack** [comment]
My 2c on this is that idle mods should be pinged. If we're asleep then it's on use to ensure we have our phones on silent really.   
Sometimes some of us will sit in idle deliberately if we're able to keep half an eye on things but can't give Homelab our full attention. That's the sort of situation where, at least personally, I would still want to be pinged.

**Comment by issy** [comment]
How do we feel about it pinging online & idle mods, and if no one is in either of those states it'll include DND mods, and then just ping all mods?

**Comment by tigattack** [comment]
I'd go for all mods if none are online/idle since anyone in DnD won't be notified, whereas people who are offline may get something through on their phone, or possibly just be invisible and not offline at all.

**Comment by issy** [comment]
Good shout, will do

File: report/report.py [hunk 1]
```diff
@@ -54,29 +45,53 @@ async def cmd_report(
         # Pre-emptively delete the message for privacy reasons
         await ctx.message.delete()
 
-        author = ctx.author
-        if author.bot:
```
Comment (kdavis) [inline report/report.py:58]: Are we no longer ignoring the bot's report?

Comment (issy) [reply report/report.py:58]: Bots don't respond to other bots. This functionality is baked into the commands module

**Review by kdavis — APPROVED** [review]

**Pull request merged**